UnitsNet/GeneratedCode/Quantities/Acceleration.g.cs

```diff
@@ -898,36 +898,88 @@ public Acceleration ToUnit(AccelerationUnit unit)
         }
 
         /// <summary>
-        ///     Converts this Acceleration to another Acceleration using the given <paramref name="unitConverter"/> with the unit representation <paramref name="unit" />.
+        ///     Converts this <see cref="Acceleration"/> to another <see cref="Acceleration"/> using the given <paramref name="unitConverter"/> with the unit representation <paramref name="unit" />.
         /// </summary>
         /// <param name="unit">The unit to convert to.</param>
         /// <param name="unitConverter">The <see cref="UnitConverter"/> to use for the conversion.</param>
         /// <returns>A Acceleration with the specified unit.</returns>
         public Acceleration ToUnit(AccelerationUnit unit, UnitConverter unitConverter)
         {
-            if (Unit == unit)
+            if (TryToUnit(unit, out var converted))
             {
-                // Already in requested units.
-                return this;
+                // Try to convert using the auto-generated conversion methods.
+                return converted!.Value;
             }
             else if (unitConverter.TryGetConversionFunction((typeof(Acceleration), Unit, typeof(Acceleration), unit), out var conversionFunction))
```

Comment:
Shouldn't extensibility conversions be attempted first?

So it is possible to override the built-in functions. I'm not sure what would be a good reason to do so though, other than duct-tape fixing bugs or precision issues. More of a gut-feeling, that extensibility should have precedence. I don't know if the extra checks would have any meaningful impact on performance though.

Reply:
Generally, I'd say yes, extensibility should have precedence. But if you went that route, you may as well abandon this PR and use as-is. 

This is purely a performance optimization as I saw the benchmark was taking 20x longer for As conversions 👎🏻 

However, this optimization would mean that you could not override the conversion functions that we define in the json files at runtime due to the if/else ordering.

I guess the question is
- Do we need performance? If so, merge this PR
- Is extensibility more important than performance? Abandon this!

Reply:
As per https://github.com/angularsen/UnitsNet/issues/1069, it seems the consensus is that the performance is now so bad that this is the better compromise.